### ISSUE TITLE
pipe is not working in the browser

### ISSUE THREAD

**Issue by seibelj** [emscripten-core/emscripten#4405]
I have my emscripten compiled JS file.

Inside of it is this function:

``` javascript
function _pipe($fd) {
 $fd = $fd|0;
 var $0 = 0, $1 = 0, $vararg_buffer = 0, label = 0, sp = 0;
 sp = STACKTOP;
 STACKTOP = STACKTOP + 16|0; if ((STACKTOP|0) >= (STACK_MAX|0)) abort();
 $vararg_buffer = sp;
 HEAP32[$vararg_buffer>>2] = $fd;
 $0 = (___syscall42(42,($vararg_buffer|0))|0);
 $1 = (___syscall_ret($0)|0);
 STACKTOP = sp;return ($1|0);
}
```

___syscall42 is right beneath ___syscall51 (the `acct` syscall), which it calls:

``` javascript
function ___syscall51(which, varargs) {if (ENVIRONMENT_IS_PTHREAD) { return _emscripten_sync_run_in_main_thread_2(138, 51, varargs) }
  SYSCALLS.varargs = varargs;
  try {
   // acct
      return -ERRNO_CODES.ENOSYS; // unsupported features
    } catch (e) {
    if (typeof FS === 'undefined' || !(e instanceof FS.ErrnoError)) abort(e);
    return -e.errno;
  }
  }function ___syscall42() {
  return ___syscall51.apply(null, arguments)
  }
```

As a result, the return to `pipe(int fd[2])` in my C code returns an error. Does pipe simply not work in a browser? Am I missing something obvious?

**Comment by juj** [comment]
Yeah, like it mentions there, the pipe syscall is unsupported. See #4378.

**Comment by seibelj** [comment]
Ahhh gotcha, thanks! Hopefully it will be merged in soon...

**Comment by atrosinenko** [comment]
PR #4378 was merged as #4935. You can try it using the `incoming` variant of SDK.
